src/constraint/encoding.ts

```diff
@@ -140,6 +140,34 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true;
     }
   },{
+    name: 'scalePropertiesSupportedByScaleType',
+    description: 'Scale properties must be supported by correct scale type',
+    properties: [Property.SCALE, Property.SCALE_TYPE, Property.TYPE],
+    allowEnumSpecForProperties: true,
+    strict: true,
+    satisfy: (encQ: EncodingQuery, schema: Schema, opt: QueryConfig) => {
+      if (encQ.scale) {
+        const scale: ScaleQuery = encQ.scale as ScaleQuery;
+        const hasScaleType: boolean = contains(keys(scale), Property.SCALE_TYPE);
```

Comment:
This is a terribly expensive way to check if `scale` has `type` property.  (O(n)!)  

Plus, it doesn't help you distinguish enumSpec from specific values.